### DISCUSSION TITLE
What's the difference between kv_service and meta_service?

### DISCUSSION THREAD

**Discussion by Xuanwo** [databendlabs/databend#5320] [Q&A] [answered]
Hi, in `common/grpc/src/client_conf.rs`, we have `MetaGrpcClientConf` like the following:

https://github.com/datafuselabs/databend/blob/7bce21c33288901659e10588749c4186cb4783a3/common/meta/grpc/src/grpc_client_conf.rs#L19-L46

It's wired that `meta_service_config` is the same as `kv_service_config`, what's the difference here? Is it possible to merge them?

> BTW, no one uses `kv_service_config` through my searching of code base.

---

Call @drmingdrmer @ariesdevil for help

**Comment by bohutang** [comment, marked as answer]
I think it's the same now.

**Reply by drmingdrmer** [reply]
The same now.

As I recall somewhere, these two have distinguished usage. Let me check it out.

**Reply by drmingdrmer** [reply]
The only difference I found is: `MetaClientProvider` supports building a local kv-client but not for a meta-client.
It seems one of them can be removed:

https://github.com/datafuselabs/databend/blob/7bce21c33288901659e10588749c4186cb4783a3/query/src/common/meta/meta_client.rs#L40-L57

**Reply by Xuanwo** [reply]
How about removing `kv_service_config` after versioned config has landed?
